hubblestack_nebula_v2/mask.yaml

```diff
@@ -1,78 +1,37 @@
-# this section contains the information about masking such objects
-blacklisted_objects:
-
-        - query_name: 'running_procs' #  Name of the osquery to be masked. Put '*' to match all queries.
-          column: 'open_files'        #  Column name in the osquery to be masked. Regex/* NOT SUPPORTED
-          attribute_to_check: 'fd'    #  In the json object, this attribute will be tested against
-                                      #  blacklisted patterns.
-          attributes_to_mask:         #  These attributes will be masked 
-            - 'path'                  #  Mask the value of path attribute
-            - 'cwd'                   #  Mask the value of cwd attribute
-          blacklisted_patterns:       #  'attribute_to_check' will be tested against these strings
-            - 'pattern1'              #   Regex NOT SUPPORTED here
-            - 'pattern2'  
-            - 'pattern3'  
-
-
-      # Examples
-      
-      # Mask ETCDCTL passwords value. 
-      # This is how it's json looks currently:  
-      #      environment: [...{ value: 'SOME_PASSWORD', variable_name: 'ETCDCTL_READ_PASSWORD' }...]
-        - query_name: 'docker_running_procs'
-          column: 'environment'
-          attribute_to_check: 'variable_name'
-          attributes_to_mask:
-            - 'value'
-          blacklisted_patterns:
-            - 'ETCDCTL_READ_PASSWORD'
-            - 'ETCDCTL_WRITE_PASSWORD'
+# Sample masking for documentation. Please see nebula_osquery.py for more
+# documentation
+#
+# Pattern that will replace whatever is masked
+mask_with: '***masked*by*hubble***'
+
+# Target and mask strings based on regex patterns
+# Can limit search specific queries and columns
+#blacklisted_strings:
+#    - query_name: 'running_procs'  # Name of the osquery to be masked.
+#                                   # Put '*' to match all queries. Note
+#                                   # that query_name doesn't support
+#                                   # full globbing. '*' is just given
+#                                   # special treatment.
+#      column: 'command_line'  # Column name in the osquery to be masked. No regex or glob support
+#      # See below for documentation of these blacklisted patterns
+#      blacklisted_patterns:
+#          - '(prefix)(password)(suffix)'
+
+# Some osquery results are formed as lists of dicts. We can mask
+# based on variable names within these dicts.
+#blacklisted_objects:
+#
+#    - query_name: 'running_procs'  # Name of the osquery to be masked.
+#                                   # Put '*' to match all queries. Note
+#                                   # that query_name doesn't support
+#                                   # full globbing. '*' is just given
+#                                   # special treatment.
+#      column: 'environment'  # Column name in the osquery to be masked. No regex or glob support
+#      attribute_to_check: 'value' # In the inner dict, this is the key
```

Comment:
I feel that you should swap line number 30 and 33. The following combo feels more intuitive than yours:
```attribute_to_check: 'env_var_name'
attribute_to_mask: 'value'```

Reply:
Agreed. I actually left the original order, but I do think your new suggestion is better.

Reply:
Er, wait, you're recommending that I change the functions of the variables? I can do that, but it's a breaking change for your existing masking. So all your masking will break next release until they upgrade.

Keep me posted if you'd like me to make that change.

(Note, if I were allowed to make breaking changes, I'd probably do this:

```
attribute_key_to_mask: 'value'
attribute_values_to_mask:
    - 'env_var_name'
    - 'path'
```

I think that's a little more descriptive.

But again, I won't make breaking changes unless both you and @sumanmehta explicitly state your your approval.